### PR TITLE
[markdown-it] Update types for markdown-it v10.0.0

### PR DESCRIPTION
**Pull request by duduluu**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

Details
- for defferent parser `Core`(lib/parser_core.js), `ParserBlock`(lib/parser_block) and `ParserInline`: their rule function use different parameters.
- for class `Token`(lib/token.js): some properties are nullable.
- better friendly for plugin developers

### CODE REVIEW

**Comment by duduluu** [comment]
@plantain-00 please review this PR

**Review by plantain-00 — APPROVED** [review]
Good on my code.

**Comment by duduluu** [comment]
Please merge this PR. plantain-00, one of the contributors, has reviewed this PR.

**Comment by uniqueiniquity** [comment]
Could an owner please review this PR?
@rapropos @bradleyayers @davidka @timjb @patsimm @ifiokjr

**Comment by duduluu** [comment]
Hi, @uniqueiniquity ,  plantain-00 is one of the authors of @types/markdown-it

File: types/markdown-it/lib/helpers/parse_link_label.d.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import State = require('../rules_core/state_core');
+
+declare function parseLinkLabel(state: State, start: number, disableNested?: boolean): number;
```
Comment (Mister-Hope) [inline types/markdown-it/lib/helpers/parse_link_label.d.ts:3]: I think there is a bug here. The `state` could also be `StateInline` here. Could you check it?

**Review by Mister-Hope — COMMENTED** [review]
I think there is a bug here

File: types/markdown-it/lib/helpers/parse_link_label.d.ts (hunk 1 above)
Comment (Mister-Hope) [reply types/markdown-it/lib/helpers/parse_link_label.d.ts:3]: @plantain-00 @duduluu

**Comment by duduluu** [comment]
@Mister-Hope 
Fixed

**Pull request merged**